### PR TITLE
memort fixes

### PR DESCRIPTION
**Pull request by ceolin**
Fix user-after-free and invalid memory access

### CODE REVIEW

File: src/lib/comms/sol-mqtt-impl-mosquitto.c [hunk 1]
```diff
@@ -274,7 +274,8 @@ sol_mqtt_on_message_wrapper(void *data)
 
     sol_mqtt_message_free(message);
 
-    mqtt->message_timeout = NULL;
+    if (mqtt)
+        mqtt->message_timeout = NULL;
```
Comment (cabelitos) [inline src/lib/comms/sol-mqtt-impl-mosquitto.c]: Why not just set to NULL?

Comment (ceolin) [inline src/lib/comms/sol-mqtt-impl-mosquitto.c]: because `mqtt` can be NULL

Comment (cabelitos) [inline src/lib/comms/sol-mqtt-impl-mosquitto.c]: ok. +1

**Comment by ceolin** [comment]
merged, thank you for reviewing it.

**Pull request closed without merging**